include/aspect/material_model/visco_plastic.h

```diff
@@ -162,7 +162,15 @@ namespace aspect
           composite
         } viscous_flow_law;
 
-
+        /**
+         * Enumeration for selecting which type of yield mechanism to use.
+         * Select between drucker prager and stress limiter.
+         */
+        enum yield_scheme
```

Comment:
Can you use CameCase for types? So here `YieldScheme` (or, if you want to match the name of the variable, `YieldMechanism`.